### ISSUE TITLE
Send emails to different segments of users

### ISSUE THREAD

**Issue by pmackay** [folklabs/openlocal#59]
Update by @djwesto on 02 Feb 2017:

CPNF’s original brief specified we must develop a way to 'segment' site members and that messages can be sent to cross sections of site users (i.e. traders or event organisers).

We suggest member segmentation should be conducted on-site, rather than with Mailchimp’s tools by way of some customised AdministrationView listings. This would enable us to use segmentation for site content, as well as on Mailchimp. We would add new “metadata” fields to user profiles to capture relevant user details (i.e. post codes).

Potential segments:
* News contributors
* What’s on event contributors
* Owners of organisation listings

@pmackay also believes we can then send mail from within Drupal, although we need to test and confirm performance of this approach.

Francis at CPNG also wants to be able to segment by _User Interest_ (e.g. Parks & Gardens) so we should also look at offering a way to segment by Content categories?

**NB.** We also need to avoid spamming users with duplicate emails if they’ve signed up to multiple groups.

---

Initial comment by @pmackay:

@GilesGibson can you give more detail what kinds of messages this could include? Please list all varieties.

Also specific info on when Mailchimp can and cannot be used. Note that one of the significant problems with emailing via the website is that it is slow. I think in past experiments it took a day to email everyone. So it would be problematic to use this for large scale mailouts.

**Comment by GilesGibson** [comment]
We need to review the pros and cons of using web site versus mailchimp for emailing subsets of users. Need to know what the issues are.

**Comment by pmackay** [comment]
It would help to have an outline of what the key kinds of messages are you want to send.

**Comment by GilesGibson** [comment]
To be honest simply formatted emails would be fine for now. Mailchimp is nice in that we can easily add pictures and the overall look and feel is better. However, we would need to figure out how to segment the mailchimp user base easily.

**Comment by pmackay** [comment]
Sorry, I'm not being clear. Please describe, for each type of message:

* What message content you need to send
* What group you would send it to
* Roughly how often might you send that kind of message

e.g.:

Newsletter
All users
Bi-weekly

---
Update on news policies
News authors
Every 6 months

**Comment by GilesGibson** [comment]
content could be update on web site, traffic, reach etc. Highlight a new feature. 
User group would be, for instance, all service listers.
probably not sent more than once a month.

**Comment by djwesto** [comment]
@GilesGibson by 'service listers' I take it you mean 'Any users who've added a Service listing'?

**Comment by GilesGibson** [comment]
yes, any user that we have set up with a listing in the directory.
